### ISSUE TITLE
Add custom recipe support

### ISSUE THREAD

**Issue by MisterXtreme** [minetest/minetest#5477]
I, and other modders would greatly appreciate custom recipe support.
Q: Why?
A: To allow a standard way for defining custom recipes, for example, right now to add an oven that gives you more bread when you put flour in it, you need hacky, non standard code to add a crafting recipe type for ovens, also meaning any crafting recipe guide mod needs me, or them to add support for my oven mod, for my recipes to even show up!

To tie into my last point, if a farming mod wants support for my crafting recipe system, it needs to specifically add support with a bunch of code, wheres with a custom recipe API, you could do minetest.getrecipetype(oven = 1), and you can detect my oven system, and use it.
"But why not use is mod loaded?!" you ask?
Simple, what if another mod implements my crafting system after I deprecate my oven mod, if it defines the same recipe type as mine, then the farming mod can just use it instead, allowing for a kind f recipe group.

Q: Why not use the code everyone else uses to add custom recipes?!"
A: Please read the full post, that will help you understand as it answers quite a few concerns

TL;DR Read the whole thing, I can't condense it anymore then it already is.

**Comment by kaeza** [comment]
Related: #1111, #1079.

**Comment by rubenwardy** [comment]
Duplicate of #1079
